Lots o' bug squashing and refactoring of Board and Game

jwehorner opened the pull request.

jwehorner committed:
Merge pull request #4 from martin509/master

update

jwehorner committed:
Lots o' bug squashing and refactoring of Board and Game

martin509 commented:
Wow that's a lot. Looks pretty nice - just for the sake of having it in text, what's being refactored specifically?

jwehorner commented:
I changed all the locations in board and game to use Point instead of two ints, fixed moving foxes, and changed game to be a decorator for board so you don't need to bypass game when moving pieces. I also created a unit test for fox for debugging purposes
